### PR TITLE
0x api docs

### PR DESCRIPTION
**Pull request by fragosti**
http://staging-0xproject.s3-website-us-east-1.amazonaws.com/docs/api

### CODE REVIEW

**Comment by chriskalani** [comment]
Nice! This is looking good!

Pretty small thing but I would use the code tags on the text left column of these tables:

Screen Shot 2019-12-21 at 10 48 28 PM

So it matches the other pages (ex core concepts):

Screen Shot 2019-12-21 at 10 48 41 PM

**Comment by chriskalani** [comment]
Also this section looks a little messed up. The code snippet and the HttpCall component and smashed up against each other: http://staging-0xproject.s3-website-us-east-1.amazonaws.com/docs/api#examples-6

**Comment by fragosti** [comment]
@chriskalani thanks, was wondering about the code tag as well. We aren't consistent about that afaik.

Regarding the code snippet in that HTTP call - that is on purpose. Some HTTP calls (POST calls) have a body. So in that case the snippet is part of the call.

File: mdx/api/index.mdx [hunk 1]
```diff
@@ -0,0 +1,776 @@
...
+If either `assetDataA` or `assetDataB` are specified, the response will only return pairs that include that asset.
+If both `assetDataA` and `assetDataB` are specified, the response will only return that pair.
+
+| Query Param  | Description                                                                                                                                                 |
+| ------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| assetDataA   | (Optional) A filter on the type of asset your may want information about in the form of [asset data](/docs/guides/v3-specification#assetdata).              |
+| assetDataB   | (Optional) A filter on the type of asset your may want information about in the form of [asset data](/docs/guides/v3-specification#assetdata).              |
+
+
+### Response
```
Comment (johnrjj) [inline mdx/api/index.mdx:458]: This was confusing to me -- it wasn't apparent enough that the response is enveloped in a pagination structure. I don't think the pagination banner is enough here. (Devs eyes just go right to request/response structures)

File: mdx/api/index.mdx [hunk 2]
```diff
@@ -0,0 +1,776 @@
...
+| ------------       | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| sellToken          | The ERC20 token address or symbol of the token you want to send. "ETH" can be provided as a valid `sellToken`.                                                                                                      |
+| buyToken           | The ERC20 token address or symbol of the token you want to receive.                                                                                                                                                 |
+| sellAmount         | (Optional) The amount of `sellToken` (in `sellToken` units) you want to send. If `sellAmount` is provided, `buyAmount` will be ignored.                                                                                           |
```
Comment (johnrjj) [inline mdx/api/index.mdx]: Got bit here -- either sellAmount or buyAmount is required (even though both say optional)

File: mdx/api/index.mdx (hunk 1 above)
Comment (johnrjj) [reply mdx/api/index.mdx:458]: (Same with the request part, need a better way to show it supports pagination)

File: mdx/api/index.mdx [hunk 3]
```diff
@@ -0,0 +1,776 @@
...
+| takerAddress       | (Optional) The address which will fill the quote. When provided the gas will be estimated and returned. An eth_call will also be performed. If this fails a Revert Error will be returned in the response.          |
+
+
+### Response
+
+| Field       | Description                                                                                                                                                              |
+| ----------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| price       | The price of 1 `buyToken` denoted in the `sellToken`. This price does not include the `slippage` provided in the request above.                                          |
+| to          | The address of the contract to send call `data` to.                                                                                                                      |
+| data        | The calldata required to be sent to the to contract address.                                                                                                             |
```
Comment (johnrjj) [inline mdx/api/index.mdx]: I would love a link/more info to what `data` does -- from the lens of a frontend dev, I don't know what this is for

File: mdx/api/index.mdx (hunk 1 above)
Comment (fragosti) [reply mdx/api/index.mdx:458]: @johnrjj how do you suggest we do that? Full pagination structure every time?

File: mdx/api/index.mdx [hunk 4]
```diff
@@ -0,0 +1,886 @@
...
+| `takerAssetAmount`      | Amount of the taker's asset(s) the maker will accept in exchange for their maker asset(s). In order to calculate the price the maker is offering, one can divide the `makerAssetAmount` by the `takerAssetAmount` (the calculation is [a bit more complex](/docs/guides/v3-specification#assetproxy) if multiple assets are involved). |
+| `expirationTimeSeconds` | Timestamp in seconds of when the order expires. Expired orders cannot be filled.                                                                                                                                                                                                                                                       |
+| `salt`                  | A value that can be used to guarentee order uniqueness. Typically it is set to a random number.                                                                                                                                                                                                                                        |
```
Comment (dekz) [inline mdx/api/index.mdx]:
```suggestion
| `salt`                  | A value that can be used to guarantee order uniqueness. Typically it is set to a random number.                                                                                                                                                                                                                                        |
```

Do we have this in other places as well

File: mdx/api/index.mdx [hunk 5]
```diff
@@ -0,0 +1,886 @@
...
+  "value": "1200000",
+  "gasPrice": "8",
```
Comment (dekz) [inline mdx/api/index.mdx]: I'd update this with the latest response, a number of fields have been changed or removed.

```json
{
  "price": "166.933760683760683761",
  "to": "0x61935cbdd02287b511119ddb11aeb42f1593b7ef",
  "data": "0x8bc8efb30000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000098968000000000000000000000000000000000000000000000000000000000000003c000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000b0dc61047847732a013ce27341228228a38655a00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000001100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000b71b0000000000000000000000000000000000000000000000000000000000000118cc0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000967a67f0ab52fa0df98c6823c36b346e4bb12576f6c02a6becded7a7572f530e7d5b97de00000000000000000000000000000000000000000000000000000000000001c000000000000000000000000000000000000000000000000000000000000002a00000000000000000000000000000000000000000000000000000000000000300000000000000000000000000000000000000000000000000000000000000030000000000000000000000000000000000000000000000000000000000000000a4dc1600f30000000000000000000000006b175474e89094c44da98b954eedeac495271d0f000000000000000000000000b0dc61047847732a013ce27341228228a38655a000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000020000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000024f47261b0000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc20000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000010400000000000000000000000000000000000000000000000000000000000000",
  "value": "1200000000000000",
  "gasPrice": "8000000000",
  "protocolFee": "1200000000000000",
  "makerAssetAmount": "10000000",
  "totalTakerAssetAmount": "59904",
  "orders": [
    {
      "chainId": 1,
      "exchangeAddress": "0x61935cbdd02287b511119ddb11aeb42f1593b7ef",
      "makerAddress": "0xb0dc61047847732a013ce27341228228a38655a0",
      "takerAddress": "0x0000000000000000000000000000000000000000",
      "feeRecipientAddress": "0x1000000000000000000000000000000000000011",
      "senderAddress": "0x0000000000000000000000000000000000000000",
      "makerAssetAmount": "12000000",
      "takerAssetAmount": "71884",
      "makerFee": "0",
      "takerFee": "0",
      "expirationTimeSeconds": "2524604400",
      "salt": "77492104380396904871793237448137396496258875317051189624210465011432762873822",
      "makerAssetData": "0xdc1600f30000000000000000000000006b175474e89094c44da98b954eedeac495271d0f000000000000000000000000b0dc61047847732a013ce27341228228a38655a000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000020000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
      "takerAssetData": "0xf47261b0000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2",
      "makerFeeAssetData": "0x",
      "takerFeeAssetData": "0x",
      "signature": "0x04"
    }
  ]
}

```

File: mdx/api/index.mdx (hunk 1 above)
Comment (johnrjj) [reply mdx/api/index.mdx:458]: no, i don't think so -- other docs show paginated records like this; i think it's just a usability thing and can be as simple as making Paginated bold within proximity of the response (I think it's a little too far up and the context gets lost) or something, the user just needs a nudge when looking at the response (some responses return pagination others don't so it's easy to forget when new/unfamiliar).

File: mdx/api/index.mdx [hunk 6]
```diff
@@ -0,0 +1,776 @@
...
+| makerFeeAssetData     | (Optional) Identifying token information of the asset(s) the maker must pay to the feeRecipientAddress.                                                                                                                                                                                                                                           |
+| takerFeeAssetData     | (Optional) Identifying token information of the asset(s) the taker must pay to the feeRecipientAddress.                                                                                                                                                                                                                                           |
+| senderAddress         | (Optional) An advanced field that doesn't need to be set. It allows the maker to enforce that the order flow through some additional logic before it can be filled (e.g., a KYC whitelist) -- more on the ability to extend 0x later.                                                                                                             |
+
+
+### Response
```
Comment (johnrjj) [inline mdx/api/index.mdx:539]: e.g. here's a place where if you control-f straight to the response, you won't see the pagination banner (on ln ~382)

**Comment by chriskalani** [comment]
@fragosti the syntax highlighting is super light. Let's darken it up.

New blue: #1F6CBC
New red: #934943

Here is the before & after:

Screen Shot 2020-01-14 at 7 09 37 PM

File: mdx/api/index.mdx (hunk 1 above)
Comment (fragosti) [reply mdx/api/index.mdx:458]: Hmmm ok. Will do some thinking.

File: mdx/api/index.mdx (hunk 6 above)
Comment (fragosti) [reply mdx/api/index.mdx:539]: The thing with this is that pagination affects both the request and the response, so I can't just stick it here. Will see how I can improve this though.

Comment (johnrjj) [reply mdx/api/index.mdx:539]: all good -- it's just one data point so i don't want to over-index on it, and definitely not a ship blocker

File: mdx/api/index.mdx (hunk 4 above)
Comment (fragosti) [reply mdx/api/index.mdx]: In core-concepts. Lots of fixes going into core-concepts...

**Pull request merged**